src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/Dp.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.Intrinsics.Arm
+{
+    /// <summary>
+    /// This class provides access to the ARMv8.2-DotProd hardware instructions via intrinsics
+    /// </summary>
+    [Intrinsic]
+    [CLSCompliant(false)]
+    public abstract class Dp : AdvSimd
+    {
+        internal Dp() { }
+
+        public static new bool IsSupported { get => IsSupported; }
+
+        [Intrinsic]
+        public new abstract class Arm64 : AdvSimd.Arm64
```

Comment:
I'm confused why this is here if there are no 64-bit intrinsics in this class.

Reply:
I added this as a placeholder and I anticipated to wait till Tanner merges his PR and update thereafter

Reply:
My understanding that Dp.Arm64.IsSupported needs to return false if Dp.IsSupported is false.
While now it returns Dp.Arm64.IsSupported true if AdvSimd.Arm64.IsSupported is true, so we need to treat Dp.Arm64.IsSupported as intrinsics instead of being inherited from AdvSimd.Arm64.

Reply:
`Dp.Arm64.IsSupported` needs to return true if `Dp.IsSupported` and we are in a 64-bit process. It being `true` implies `Dp.IsSupported` and `ArmBase.Arm64.IsSupported` (which each also implies `ArmBase.IsSupported`).

Having it recursive here and adding the relevant entry to `Arm64VersionOfIsa` should hook it all up correctly (same as was done for Rdm: https://github.com/dotnet/runtime/pull/38957/files#diff-8e07abb3dff9bfe2e051a071c787f18bR27)

It exists because of https://github.com/dotnet/runtime/issues/34587 (and is the same reason as https://github.com/dotnet/runtime/pull/38460 was created), which is that we otherwise get into weird scenarios where `Dp.Arm64.IsSupported` can return true (because it actually resolves to `ArmBase.Arm64.IsSupported` when `Dp.IsSupported` returns false.

Reply:
@CarolEidt Do you have any other questions here?

Reply:
No, makes sense (sort of!)